yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -165,31 +167,31 @@ template <
     typename D,
     typename F,
     typename = typename std::enable_if<std::is_callable<F(U), D>::value>::type>
-class MapOperator : public ObservableOperator<U, D> {
+class MapOperator : public ObservableOperator<U, D, MapOperator<U, D, F>> {
+  using ThisOperatorT = MapOperator<U, D, F>;
+  using Super = ObservableOperator<U, D, ThisOperatorT>;
```

Comment:
nit: As we already shorten some long declarations with `using` keyword, we might utilize that for members of the Super class also. `Super::` is repeating too many times. Put this to 
```
using Super::upstream_;
```